### PR TITLE
[website] 404 page and other route considerations

### LINKED ISSUE #547: [Website] Add 404 page
As a viewer of the site, I need a proper, helpful 404 page, so that I'm not confused or frustrated by a blank screen on invalid routes.

#### AC
1. Invalid routes redirect to 404 page
2. Valid, but currently broken, routes like /components do something useful
3. 404 page itself has considered design and content

#### Other Considerations
- See this comment: https://github.com/mineral-ui/mineral-ui/pull/539#discussion_r164145155

### PR DESCRIPTION
**Pull request by kylegach**
### Description

- Add 404 page route
- Add redirects for cases like /components/not-a-route & /components/box/not-an-example

Note: We looked at Netlify to see what it would take to [add a custom 404 server-side](https://www.netlify.com/docs/redirects/#custom-404). Since that would require generating a static 404.html file, we decided the JS-dependent version was fine.

### Motivation and context

Closes #547

### Screenshots, videos, or demo, if appropriate

- https://547-404-page--mineral-ui.netlify.com/404
- https://547-404-page--mineral-ui.netlify.com/components
- https://547-404-page--mineral-ui.netlify.com/components/not-a-component
- https://547-404-page--mineral-ui.netlify.com/components/button/not-an-example

### How to test

1. For the four links above:
    1. The first should show a fun 404
    1. The second should redirect to avatar (the first component)
    1. The third should redirect to avatar (the first component)
    1. The fourth should redirect to button

### Types of changes

- Other (provide details below)

Just website stuff

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing [n/a]
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered [n/a]
* [x] Documentation created or updated [n/a]
* [x] Tested in [sandbox](https://github.com/facebookincubator/create-react-app#getting-started) if new component or breaking change [n/a]

### How does this PR make you feel?

glitchy cat

### CODE REVIEW

File: src/website/app/Router.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ import ComponentDocExample from './ComponentDocExample';
 import Page from './Page';
 import sections from './pages';
 import ComponentDoc from './pages/ComponentDoc';
+import FourOhFour from './pages/FourOhFour';
```
Comment (brentertz) [inline src/website/app/Router.js]: How about `PageNotFound` or `FileNotFound`?  I'd also rename the `404.md` file name match as well.

File: src/website/app/Router.js [hunk 2]
```diff
@@ -129,7 +134,15 @@ export default function Router({ demoRoutes }: Props) {
...
         path="/components/:componentId"
         render={({ location, match }) => {
-          const componentId = match.params.componentId || firstDemoSlug;
+          const componentId =
+            flatDemoRoutes.find(
+              route => route.slug === match.params.componentId
+            ) && match.params.componentId;
+
+          if (!componentId) {
+            return <Redirect to={`/components/${firstDemoSlug}`} />;
```
Comment (brentertz) [inline src/website/app/Router.js]: I am assuming that you weighed the benefit an incorrect component route going to the 404 page vs. it going to the first component and decided that the latter was better.  Is that correct?

File: src/website/app/pages/FourOhFour/index.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+};
+
+const styles = {
+  floatingMineralsContainer: {
+    filter: 'hue-rotate(150deg) contrast(150%)',
+    margin: '1em auto',
+    maxWidth: '25em',
+    overflow: 'hidden',
+    position: 'relative',
+    transform: 'rotate(180deg)',
```
Comment (brentertz) [inline src/website/app/pages/FourOhFour/index.js]: My opinion is that leaving the image upright is better.  I did not even recognize what it was when flipped.

File: src/website/app/Router.js (hunk 2 above)
Comment (kylegach) [reply src/website/app/Router.js]: Yep. Seemed like an opinionated redirect was more helpful.

**Pull request merged**